addons/account/models/partner.py

```diff
@@ -514,6 +514,17 @@ def create(self, vals_list):
                     vals['supplier_rank'] = 1
         return super().create(vals_list)
 
+    def unlink(self):
+        """
+        Prevent the deletion of a partner "Individual", child of a company if:
+        - partner in 'account.move'
+        - state: all states (draft and posted)
+        """
+        moves = self.sudo().env['account.move'].search_count([('partner_id', 'in', self.ids), ('state', 'in', ['draft', 'posted'])])
+        if moves:
+            raise UserError(_("Record cannot be deleted. Partner used in Accounting"))
```

Comment:
A `RedirectWarning` to the list view of `account.move` filtered on the partner would be great (although it might be weird because you can't have the `sudo`, but it is better like that anyways)